src/color.rs

```diff
@@ -314,18 +373,24 @@ impl<T: Primitive> IndexMut<usize> for $ident<T> {
     }
 }
 
+impl<T: Primitive + 'static> From<[T; $channels]> for $ident<T> {
+    fn from(c: [T; $channels]) -> Self {
+        Self(c)
+    }
+}
+
 )* // END Structure definitions
 
     }
 }
 
 define_colors! {
-    Rgb, 3, 0, "RGB", ColorType::Rgb8, #[doc = "RGB colors"];
-    Bgr, 3, 0, "BGR", ColorType::Bgr8, #[doc = "BGR colors"];
-    Luma, 1, 0, "Y", ColorType::L8, #[doc = "Grayscale colors"];
-    Rgba, 4, 1, "RGBA", ColorType::Rgba8, #[doc = "RGB colors + alpha channel"];
-    Bgra, 4, 1, "BGRA", ColorType::Bgra8, #[doc = "BGR colors + alpha channel"];
-    LumaA, 2, 1, "YA", ColorType::La8, #[doc = "Grayscale colors + alpha channel"];
+    Rgb, 3, 0, "RGB", ChannelsType::Rgb, #[doc = "RGB colors"];
+    Bgr, 3, 0, "BGR", ChannelsType::Bgr, #[doc = "BGR colors"];
+    Luma, 1, 0, "Y", ChannelsType::L, #[doc = "Grayscale colors"];
+    Rgba, 4, 1, "RGBA", ChannelsType::Rgba, #[doc = "RGB colors + alpha channel"];
+    Bgra, 4, 1, "BGRA", ChannelsType::Bgra, #[doc = "BGR colors + alpha channel"];
+    LumaA, 2, 1, "YA", ChannelsType::La, #[doc = "Grayscale colors + alpha channel"];
```

Comment:
This and Pixel::CHANNELS_TYPE seems to be the only place that ChannelsType is actually used. Rather than adding an entire extra type to the public API for these, could you instead just pass the 8 and 16 bit versions of each ColorType as two macro arguments? Then color_type() could select between them based on the type argument

Reply:
This could be removed, and if that's the preference for now I will so to merge, but I think there needs to be some decision about the ability to abstract logic over channels without having to abstract it over the partial cartesian product of channels and bit depths that `ColorType` represents, as discussed in the `color_type` review comment.

Reply:
Codifying that ColorType represents a Cartesian product of channels and bit depths is what concerns me. Currently the only promise we make is that pixels of any color type are an integer number of bytes. ExtendedColorType make the even weaker promise that all pixels are the same number of bits. 

To give examples of where this would break down, some pixel formats might use different numbers of bits for the RGB channels as they do for the alpha channel (i.e. R10G10B10A2). Or they might even have the number of bits per channel differ between different color channels (R5G6B5).
 
Perhaps we want to rule out these formats for ColorType, but I'd rather make that decision separately from the rest of this PR. (We probably do want to support weird formats like these for ExtendedColorType, the entire point of that enum is to encode weird formats we want to be able to name without claiming support for operating on them throughout the library)

Reply:
I agree with this, but see it as more argument in support having separate types to express what channels are in a pixel/image's color space abstracted from their depth, color space and encoding. However, I can see that this is something that can be dealt with independently from this PR, so have removed `ChannelsType`.

Reply:
Separate PR sounds good, let's not mix feature and bugfix changes if we can avoid them.